### PR TITLE
NameByIdLoader: Fetch names for every store_id

### PR DESCRIPTION
**Pull request by clemens-msupply**
Prevents potential bug where names from different stores are requested.

I did this because I thought I ran into this bug but that wasn't the case. Thought the change would be useful anyway.

### CODE REVIEW

File: graphql/core/src/loader/name.rs [hunk 1]
```diff
@@ -37,26 +37,31 @@ impl Loader<NameByIdLoaderInput> for NameByIdLoader {
...
-            .map_err(StandardGraphqlError::from_list_error)?;
-
-        Ok(names
-            .rows
-            .into_iter()
-            .map(|name| (NameByIdLoaderInput::new(store_id, &name.name_row.id), name))
-            .collect())
+        // store_id -> Vec of name_id
+        let mut store_name_map = HashMap::<String, Vec<String>>::new();
+        for item in ids_with_store_id {
```
Comment (andreievg) [inline graphql/core/src/loader/name.rs:42]: IdPair is pretty ambiguous data type, i tried to make it a bit cleaner with implementations, i think it's a good idea to add store_id and name_id getters to NameByIdLoaderInput, and use then here

File: graphql/core/src/loader/name.rs [hunk 2]
```diff
@@ -37,26 +37,31 @@ impl Loader<NameByIdLoaderInput> for NameByIdLoader {
...
+        }
+        let mut output = HashMap::<NameByIdLoaderInput, Self::Value>::new();
+        for (store_id, names) in store_name_map {
+            let names = self
+                .service_provider
+                .general_service
+                .get_names(
+                    &service_context,
+                    &store_id,
+                    None, // TODO this needs to be ALL without limit
```
Comment (andreievg) [inline graphql/core/src/loader/name.rs:56]: yep, agree with another comment you've made in default context, I think we need a way to not use pagination in services getters. Either with another enum variant for service pagination input or do pagination validation in graphql

**Review by andreievg — APPROVED** [review]
Thanks, looks good

File: graphql/core/src/loader/name.rs (hunk 1 above)
Comment (clemens-msupply) [reply graphql/core/src/loader/name.rs:42]: You mean to get the `HashMap<store_id, Vec<name_id>>`? yes can to in next iteration

**Pull request merged**